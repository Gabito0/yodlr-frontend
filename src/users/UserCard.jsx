import React from "react";
import { Link } from "react-router-dom";

/** Show limited information about a user
 *
 *  Is rendered by UserList to show a "card" for each user.
 *
 *  UserList -> UserCard
 */

const UserCard = ({ id, email, firstName, lastName, state }) => {
  return (
    <Link
      className="UserCard card mb-3 text-decoration-none"
      to={`/user/${id}`}
    >
      <div className="card-body">
        <h6 className="card-title">{email}</h6>
        <p className="card-text mb-1">
          <strong>First Name:</strong> {firstName}
        </p>
        <p className="card-text">
          <strong>Last Name:</strong> {lastName}
        </p>
        <span
          className={`badge rounded-pill ${
            state === "active" ? "text-bg-success" : "text-bg-warning"
          }`}
        >
          {state}
        </span>
      </div>
    </Link>
  );
};

export default UserCard;
