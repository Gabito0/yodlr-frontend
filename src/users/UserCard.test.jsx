import React from "react";
import { render } from "@testing-library/react";
import UserCard from "./UserCard";
import { MemoryRouter } from "react-router";

it("matches snapshot with logo", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserCard
        email="testusers"
        firstName="testName"
        lastName="testLast"
        state="active"
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot with state pending", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserCard
        email="testUser2"
        firstName="testFirst"
        lastName="testLast"
        state="pending"
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
