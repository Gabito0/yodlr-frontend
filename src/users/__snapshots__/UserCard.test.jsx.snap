// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <a
    class="UserCard card mb-3 text-decoration-none"
    href="/user/undefined"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      >
        testusers
      </h6>
      <p
        class="card-text mb-1"
      >
        <strong>
          First Name:
        </strong>
         testName
      </p>
      <p
        class="card-text"
      >
        <strong>
          Last Name:
        </strong>
         testLast
      </p>
      <span
        class="badge rounded-pill text-bg-success"
      >
        active
      </span>
    </div>
  </a>
</DocumentFragment>
`;

exports[`matches snapshot with state pending 1`] = `
<DocumentFragment>
  <a
    class="UserCard card mb-3 text-decoration-none"
    href="/user/undefined"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      >
        testUser2
      </h6>
      <p
        class="card-text mb-1"
      >
        <strong>
          First Name:
        </strong>
         testFirst
      </p>
      <p
        class="card-text"
      >
        <strong>
          Last Name:
        </strong>
         testLast
      </p>
      <span
        class="badge rounded-pill text-bg-warning"
      >
        pending
      </span>
    </div>
  </a>
</DocumentFragment>
`;
